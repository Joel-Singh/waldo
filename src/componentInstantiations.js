import MapPreview from "./components/MapPreview";
import Gamescreen, { createCharacter } from "./components/Gamescreen";

import mazeWaldo from "./assets/characters/maze/waldo.jpg";
import birdPerson from "./assets/characters/maze/bird_person.jpg";
import yellowHairPerson from "./assets/characters/maze/yellowed_hair_person.jpg";

import beachWaldo from "./assets/characters/beach/waldo.jpg";
import shovelBoy from "./assets/characters/beach/shovel_boy.jpg";
import dog from "./assets/characters/beach/dog.jpg";

import snowWaldo from "./assets/characters/snow/waldo.jpg";
import ouch from "./assets/characters/snow/ouch.jpg";
import monster from "./assets/characters/snow/monster.jpg";

import mazeMap from "./assets/maps/maze.jpg";
import beachMap from "./assets/maps/beach.jpg";
import snowMap from "./assets/maps/snow.jpg";

const getMapPreviews = () => ({
  mazeMapPreview: (
    <MapPreview
      mapName="Maze"
      mapImage={mazeMap}
      mapPath="maze"
      previewCharacterInformation={[
        {
          name: "Waldo",
          image: mazeWaldo,
        },
        {
          name: "Bird Person",
          image: birdPerson,
        },
        {
          name: "Yellow Hair Person",
          image: yellowHairPerson,
        },
      ]}
    />
  ),
  beachMapPreview: (
    <MapPreview
      mapName="Beach"
      mapImage={beachMap}
      mapPath="beach"
      previewCharacterInformation={[
        {
          name: "Waldo",
          image: beachWaldo,
        },
        {
          name: "Shovel Boy",
          image: shovelBoy,
        },
        {
          name: "Dog",
          image: dog,
        },
      ]}
    />
  ),
  snowMapPreview: (
    <MapPreview
      mapName="Snow"
      mapImage={snowMap}
      mapPath="snow"
      previewCharacterInformation={[
        {
          name: "Waldo",
          image: snowWaldo,
        },
        {
          name: "Ouch",
          image: ouch,
        },
        {
          name: "Monster",
          image: monster,
        },
      ]}
    />
  ),
});

const getGamescreens = () => ({
  maze: (
    <Gamescreen
      img={mazeMap}
      characters={[
        createCharacter("Waldo", "mazeWaldo", mazeWaldo),
        createCharacter("Bird Person", "birdPerson", birdPerson),
        createCharacter(
          "Yellow Hair Person",
          "yellowHairPerson",
          yellowHairPerson
        ),
      ]}
    />
  ),
  beach: (
    <Gamescreen
      img={beachMap}
      characters={[
        createCharacter("Waldo", "beachWaldo", beachWaldo),
        createCharacter("Shovel Boy", "shovelBoy", shovelBoy),
        createCharacter("Dog", "dog", dog),
      ]}
    />
  ),
  snow: (
    <Gamescreen
      img={snowMap}
      characters={[
        createCharacter("Waldo", "snowWaldo", snowWaldo),
        createCharacter("Ouch!", "ouch", shovelBoy),
        createCharacter("Monster", "monster", monster),
      ]}
    />
  ),
});

export { getMapPreviews, getGamescreens };
