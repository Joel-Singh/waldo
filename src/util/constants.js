import mazeWaldo from "../assets/characters/maze/waldo.jpg";
import birdPerson from "../assets/characters/maze/bird_person.jpg";
import yellowHairPerson from "../assets/characters/maze/yellowed_hair_person.jpg";

import beachWaldo from "../assets/characters/beach/waldo.jpg";
import shovelBoy from "../assets/characters/beach/shovel_boy.jpg";
import dog from "../assets/characters/beach/dog.jpg";

import snowWaldo from "../assets/characters/snow/waldo.jpg";
import ouch from "../assets/characters/snow/ouch.jpg";
import monster from "../assets/characters/snow/monster.jpg";

export const characterInformation = {
  beach: [
    {
      databaseName: "beachWaldo",
      image: beachWaldo,
      displayName: "Waldo",
      coords: { x: 1286, y: 1637 },
    },
    {
      databaseName: "shovelBoy",
      image: shovelBoy,
      displayName: "Shovel Boy",
      coords: { x: 2725, y: 1090 },
    },
    {
      databaseName: "dog",
      image: dog,
      displayName: "Dog",
      coords: { x: 2153, y: 576 },
    },
  ],

  maze: [
    {
      databaseName: "mazeWaldo",
      image: mazeWaldo,
      displayName: "Waldo",
      coords: { x: 1377, y: 653 },
    },
    {
      databaseName: "birdPerson",
      image: birdPerson,
      displayName: "Bird Person",
      coords: { x: 1435, y: 687 },
    },
    {
      databaseName: "yellowHairPerson",
      image: yellowHairPerson,
      displayName: "Yellow Hair Person",
      coords: { x: 1929, y: 209 },
    },
  ],

  snow: [
    {
      databaseName: "snowWaldo",
      image: snowWaldo,
      displayName: "Waldo",
      coords: { x: 2561, y: 1419 },
    },
    {
      databaseName: "ouch",
      image: ouch,
      displayName: "Ouch!",
      coords: { x: 2313, y: 1087 },
    },
    {
      databaseName: "monster",
      image: monster,
      displayName: "Monster",
      coords: { x: 2798, y: 224 },
    },
  ],
};

export const flattenedCharacterInformation = (() => {
  const flattenedCharacterInformation = [];
  for (const map in characterInformation) {
    characterInformation[map].forEach((characterInfo) => {
      flattenedCharacterInformation.push(cloneObj(characterInfo));
    });
  }

  function cloneObj(obj) {
    return JSON.parse(JSON.stringify(obj))
  }
  return flattenedCharacterInformation;
})();

export const CHOOSING_CHARACTER_TOLERANCE = 40;
export const decisecondToMs = 100;

export function exposeCharacterToleranceAsCssVariable() {
  const root = document.documentElement;
  root.style.setProperty(
    "--choosing-character-tolerance",
    CHOOSING_CHARACTER_TOLERANCE
  );
}
