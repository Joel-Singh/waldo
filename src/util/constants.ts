import mazeWaldo from "../assets/characters/maze/waldo.jpg";
import birdPerson from "../assets/characters/maze/bird_person.jpg";
import yellowHairPerson from "../assets/characters/maze/yellowed_hair_person.jpg";

import beachWaldo from "../assets/characters/beach/waldo.jpg";
import shovelBoy from "../assets/characters/beach/shovel_boy.jpg";
import dog from "../assets/characters/beach/dog.jpg";

import snowWaldo from "../assets/characters/snow/waldo.jpg";
import ouch from "../assets/characters/snow/ouch.jpg";
import monster from "../assets/characters/snow/monster.jpg";


export const allMaps: Array<MapName> = ["maze", "beach", "snow"];
export const decisecondToMs = 100;

export const CHOOSING_CHARACTER_TOLERANCE = 40;
export function exposeCharacterToleranceAsCssVariable() {
  const root = document.documentElement;
  root.style.setProperty(
    "--choosing-character-tolerance",
    String(CHOOSING_CHARACTER_TOLERANCE)
  );
}

export type MapName = "beach" | "maze" | "snow"

interface Character {
  databaseName: string;
  image: string;
  displayName: string;
  coords: {
    x: number;
    y: number;
  };
}

export const characterInformation: Record<MapName, Character[]> = {
  beach: [
    {
      databaseName: "beachWaldo",
      image: beachWaldo,
      displayName: "Waldo",
      coords: { x: 1286, y: 1637 },
    },
    {
      databaseName: "shovelBoy",
      image: shovelBoy,
      displayName: "Shovel Boy",
      coords: { x: 2725, y: 1090 },
    },
    {
      databaseName: "dog",
      image: dog,
      displayName: "Dog",
      coords: { x: 2153, y: 576 },
    },
  ],

  maze: [
    {
      databaseName: "mazeWaldo",
      image: mazeWaldo,
      displayName: "Waldo",
      coords: { x: 1377, y: 653 },
    },
    {
      databaseName: "birdPerson",
      image: birdPerson,
      displayName: "Bird Person",
      coords: { x: 1435, y: 687 },
    },
    {
      databaseName: "yellowHairPerson",
      image: yellowHairPerson,
      displayName: "Yellow Hair Person",
      coords: { x: 1929, y: 209 },
    },
  ],

  snow: [
    {
      databaseName: "snowWaldo",
      image: snowWaldo,
      displayName: "Waldo",
      coords: { x: 2561, y: 1419 },
    },
    {
      databaseName: "ouch",
      image: ouch,
      displayName: "Ouch!",
      coords: { x: 2313, y: 1087 },
    },
    {
      databaseName: "monster",
      image: monster,
      displayName: "Monster",
      coords: { x: 2798, y: 224 },
    },
  ],
};

export const flattenedCharacterInformation = (() => {
  let flattenedCharacterInformation: Array<Character> = [];
  for (const map in characterInformation) {
    characterInformation[map].forEach((characterInfo: Character) => {
      flattenedCharacterInformation.push(cloneObj(characterInfo) as Character);
    });
  }

  function cloneObj(obj: Object) {
    return JSON.parse(JSON.stringify(obj)) as Object;
  }
  return flattenedCharacterInformation;
})();
