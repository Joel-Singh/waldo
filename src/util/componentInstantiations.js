import MapPreview from "../components/MapPreview";
import Gamescreen from "../components/Gamescreen";
import createCharacter from "./createCharacter";

import mazeMap from "../assets/maps/maze.jpg";
import beachMap from "../assets/maps/beach.jpg";
import snowMap from "../assets/maps/snow.jpg";
import { characterInformation } from "./constants";

const getMapPreviews = () => ({
  mazeMapPreview: (
    <MapPreview
      mapName="Maze"
      mapImage={mazeMap}
      mapPath="maze"
      previewCharacterInformation={parsePreviewCharInfoFromCharInfo("maze")}
    />
  ),
  beachMapPreview: (
    <MapPreview
      mapName="Beach"
      mapImage={beachMap}
      mapPath="beach"
      previewCharacterInformation={parsePreviewCharInfoFromCharInfo("beach")}
    />
  ),
  snowMapPreview: (
    <MapPreview
      mapName="Snow"
      mapImage={snowMap}
      mapPath="snow"
      previewCharacterInformation={parsePreviewCharInfoFromCharInfo("snow")}
    />
  ),
});

function parsePreviewCharInfoFromCharInfo(map) {
  return characterInformation[map].map(({ displayName, image }) => ({
    name: displayName,
    image,
  }));
}

const getGamescreens = (onAllCharactersFound) => ({
  maze: (
    <Gamescreen
      img={mazeMap}
      onAllCharactersFound={onAllCharactersFound}
      characters={parseGamescreenCharactersFromCharInfo('maze')}
    />
  ),
  beach: (
    <Gamescreen
      img={beachMap}
      onAllCharactersFound={onAllCharactersFound}
      characters={parseGamescreenCharactersFromCharInfo('beach')}
    />
  ),
  snow: (
    <Gamescreen
      img={snowMap}
      onAllCharactersFound={onAllCharactersFound}
      characters={parseGamescreenCharactersFromCharInfo('snow')}
    />
  ),
});

function parseGamescreenCharactersFromCharInfo(map) {
  return characterInformation[map].map(({ displayName, databaseName, image }) =>
    createCharacter(displayName, databaseName, image)
  );
}

export { getMapPreviews, getGamescreens };
