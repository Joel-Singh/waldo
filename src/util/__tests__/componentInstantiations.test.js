import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { getGamescreens, getMapPreviews } from "../componentInstantiations";
import { allMaps } from "../constants";

function snapshotTestComponent(component, testName) {
  test(testName, () => {
    const { container } = render(<BrowserRouter>{component}</BrowserRouter>);
    expect(container).toMatchSnapshot();
  });
}

getComponentsToSnapshotTest().forEach(({ component, name }) =>
  snapshotTestComponent(component, name)
);

function getComponentsToSnapshotTest() {
  const componentsToSnapshotTest = [
    ...allMaps.map((map) => {
      return {
        component: getMapPreviews()[map],
        name: `${map} Map Preview`,
      };
    }),
    ...allMaps.map((map) => ({
      component: getGamescreens()[map],
      name: `${map} Gamescreen`,
    })),
  ];

  return componentsToSnapshotTest;
}
