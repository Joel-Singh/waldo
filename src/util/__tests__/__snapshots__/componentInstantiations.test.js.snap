// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beach Gamescreen 1`] = `
<div>
  <div
    class="gamescreen"
    data-testid="gamescreen"
  >
    <img
      alt="Map"
      class="gamescreen__map"
      src="beach.jpg"
    />
    <div
      class="characters-overlay"
    >
      <img
        alt="Waldo"
        src="waldo.jpg"
      />
      <img
        alt="Shovel Boy"
        src="shovel_boy.jpg"
      />
      <img
        alt="Dog"
        src="dog.jpg"
      />
    </div>
    <div
      class="character-picker --positioned-with-x-y"
      data-testid="character-picker"
      style="--x: 0; --y: 0;"
    >
      <button
        type="button"
      >
        Waldo
      </button>
      <button
        type="button"
      >
        Shovel Boy
      </button>
      <button
        type="button"
      >
        Dog
      </button>
    </div>
    <div
      class="cursor-overlay cursor-overlay--visible"
      data-testid="cursor-overlay"
      style="--x: 0; --y: 0;"
    />
    <div
      class="stopwatch"
    >
      0.0
    </div>
  </div>
</div>
`;

exports[`Beach Map Preview 1`] = `
<div>
  <div
    class="map-preview"
  >
    <div
      aria-label="title"
      class="map-preview__title"
    >
      Beach
    </div>
    <img
      alt="Beach map"
      class="map-preview__map-image"
      src="beach.jpg"
    />
    <div
      class="map-preview__character-container"
      data-testid="previewCharactersContainer"
    >
      <div
        class="preview-character"
      >
        <img
          alt="Waldo"
          src="waldo.jpg"
        />
        <div
          class="preview-character__name"
        >
          Waldo
        </div>
      </div>
      <div
        class="preview-character"
      >
        <img
          alt="Shovel Boy"
          src="shovel_boy.jpg"
        />
        <div
          class="preview-character__name"
        >
          Shovel Boy
        </div>
      </div>
      <div
        class="preview-character"
      >
        <img
          alt="Dog"
          src="dog.jpg"
        />
        <div
          class="preview-character__name"
        >
          Dog
        </div>
      </div>
    </div>
    <a
      href="/beach"
    >
      <button
        class="map-preview__start-btn"
        type="buton"
      >
        START
      </button>
    </a>
  </div>
</div>
`;

exports[`Maze Gamescreen 1`] = `
<div>
  <div
    class="gamescreen"
    data-testid="gamescreen"
  >
    <img
      alt="Map"
      class="gamescreen__map"
      src="maze.jpg"
    />
    <div
      class="characters-overlay"
    >
      <img
        alt="Waldo"
        src="waldo.jpg"
      />
      <img
        alt="Bird Person"
        src="bird_person.jpg"
      />
      <img
        alt="Yellow Hair Person"
        src="yellowed_hair_person.jpg"
      />
    </div>
    <div
      class="character-picker --positioned-with-x-y"
      data-testid="character-picker"
      style="--x: 0; --y: 0;"
    >
      <button
        type="button"
      >
        Waldo
      </button>
      <button
        type="button"
      >
        Bird Person
      </button>
      <button
        type="button"
      >
        Yellow Hair Person
      </button>
    </div>
    <div
      class="cursor-overlay cursor-overlay--visible"
      data-testid="cursor-overlay"
      style="--x: 0; --y: 0;"
    />
    <div
      class="stopwatch"
    >
      0.0
    </div>
  </div>
</div>
`;

exports[`Maze Map Preview 1`] = `
<div>
  <div
    class="map-preview"
  >
    <div
      aria-label="title"
      class="map-preview__title"
    >
      Maze
    </div>
    <img
      alt="Maze map"
      class="map-preview__map-image"
      src="maze.jpg"
    />
    <div
      class="map-preview__character-container"
      data-testid="previewCharactersContainer"
    >
      <div
        class="preview-character"
      >
        <img
          alt="Waldo"
          src="waldo.jpg"
        />
        <div
          class="preview-character__name"
        >
          Waldo
        </div>
      </div>
      <div
        class="preview-character"
      >
        <img
          alt="Bird Person"
          src="bird_person.jpg"
        />
        <div
          class="preview-character__name"
        >
          Bird Person
        </div>
      </div>
      <div
        class="preview-character"
      >
        <img
          alt="Yellow Hair Person"
          src="yellowed_hair_person.jpg"
        />
        <div
          class="preview-character__name"
        >
          Yellow Hair Person
        </div>
      </div>
    </div>
    <a
      href="/maze"
    >
      <button
        class="map-preview__start-btn"
        type="buton"
      >
        START
      </button>
    </a>
  </div>
</div>
`;

exports[`Snow Gamescreen 1`] = `
<div>
  <div
    class="gamescreen"
    data-testid="gamescreen"
  >
    <img
      alt="Map"
      class="gamescreen__map"
      src="snow.jpg"
    />
    <div
      class="characters-overlay"
    >
      <img
        alt="Waldo"
        src="waldo.jpg"
      />
      <img
        alt="Ouch!"
        src="ouch.jpg"
      />
      <img
        alt="Monster"
        src="monster.jpg"
      />
    </div>
    <div
      class="character-picker --positioned-with-x-y"
      data-testid="character-picker"
      style="--x: 0; --y: 0;"
    >
      <button
        type="button"
      >
        Waldo
      </button>
      <button
        type="button"
      >
        Ouch!
      </button>
      <button
        type="button"
      >
        Monster
      </button>
    </div>
    <div
      class="cursor-overlay cursor-overlay--visible"
      data-testid="cursor-overlay"
      style="--x: 0; --y: 0;"
    />
    <div
      class="stopwatch"
    >
      0.0
    </div>
  </div>
</div>
`;

exports[`Snow Map Preview 1`] = `
<div>
  <div
    class="map-preview"
  >
    <div
      aria-label="title"
      class="map-preview__title"
    >
      Snow
    </div>
    <img
      alt="Snow map"
      class="map-preview__map-image"
      src="snow.jpg"
    />
    <div
      class="map-preview__character-container"
      data-testid="previewCharactersContainer"
    >
      <div
        class="preview-character"
      >
        <img
          alt="Waldo"
          src="waldo.jpg"
        />
        <div
          class="preview-character__name"
        >
          Waldo
        </div>
      </div>
      <div
        class="preview-character"
      >
        <img
          alt="Ouch!"
          src="ouch.jpg"
        />
        <div
          class="preview-character__name"
        >
          Ouch!
        </div>
      </div>
      <div
        class="preview-character"
      >
        <img
          alt="Monster"
          src="monster.jpg"
        />
        <div
          class="preview-character__name"
        >
          Monster
        </div>
      </div>
    </div>
    <a
      href="/snow"
    >
      <button
        class="map-preview__start-btn"
        type="buton"
      >
        START
      </button>
    </a>
  </div>
</div>
`;
